environment.js

```diff
@@ -0,0 +1,26 @@
+"use strict"
+
+function checkEnvironmentForConfig(config:Object) : Object {
+  let mentionBotEnvConfig;
+
+  try {
+    mentionBotEnvConfig = JSON.parse(process.env.MENTION_BOT_CONFIG);
+  } catch(e) {
+    mentionBotEnvConfig = {};
+  }
+
+  return Object.keys(config).reduce((previousValue, key) => {
+    let defaultConfigValue = config[key];
+
+    let environmentVariable = mentionBotEnvConfig[key];
+    let configElement = {};
+    configElement[key] = environmentVariable === undefined ? defaultConfigValue
+      : environmentVariable;
+
+    return {...previousValue, ...configElement};
```

Comment:
This may be "idiomatic" but this is extremely inefficient. For every single key you are copying the entire object, making the operation n^2 for no good reason.